syntax = "proto3";

import "google/protobuf/empty.proto";

option java_multiple_files = true;
option java_package = "frc.team852.DeepSpaceRobot";
option java_outer_classname = "Proto";

package utils;

service OpenCVInfo {
    rpc SendFrameSize (FrameSize) returns (google.protobuf.Empty) {}
    rpc SendCVData (CVData) returns (google.protobuf.Empty) {}
    rpc SendBall (Hatch) returns (google.protobuf.Empty) {
    }
    rpc SendHatch (Ball) returns (google.protobuf.Empty) {
    }
}

message FrameSize {
    double x = 1;
    double y = 2;
}

message Hatch {
    Point centroid = 1;
    double diameter = 2;
}

message Ball {
    Point centroid = 1;
    double diameter = 2;
}

message CVData {
    ReflTape left = 1;
    ReflTape right = 2;
    GaffeTape tape = 3;
}

message ReflTape {
    Point topInside = 1;
    Point centroid = 2;
    Point bottomOutside = 3;
    double degrees = 4;
    double size = 5;
}

message GaffeTape {
    Point front = 1;
    Point centroid = 2;
    Point back = 3;
    double degrees = 4;
}

message Point {
    int32 x = 1;
    int32 y = 2;
}

